dydx-stark/src/config.ts

```diff
@@ -0,0 +1,37 @@
+import types from '@chainlink/types'
+import { Requester, logger } from '@chainlink/external-adapter'
+import { util } from '@chainlink/ea-bootstrap'
+import { send } from './endpoint'
+
+export const DEFAULT_DATA_PATH = 'result'
+export const DEFAULT_ENDPOINT = send.NAME
+
+export const DEFAULT_API_ENDPOINT = 'https://api.stage.dydx.exchange/v3/price'
+
+const ENV_PRIVATE_KEY = 'PRIVATE_KEY'
+const ENV_STARK_MESSAGE = 'STARK_MESSAGE'
+const ENV_ORACLE_NAME = 'ORACLE_NAME'
+
+export type Config = types.Config & {
+  privateKey: string
+  starkMessage: string
+  oracleName: string
+}
+
+export const makeConfig = (prefix?: string): Config => {
+  const defaultConfig = Requester.getDefaultConfig(prefix)
+  defaultConfig.api.baseURL = defaultConfig.api.baseURL || DEFAULT_API_ENDPOINT
+  return {
+    ...defaultConfig,
+    privateKey: util.getRequiredEnv(ENV_PRIVATE_KEY, prefix),
```

Comment:
Are these required environment variables or optional?

Reply:
These are required.

But this adapter does not require an API_KEY, even in production; @emmick4 that feature where we now require an API_KEY for every adapter is posing a problem.

Reply:
yep running into that now updating adapters to typescript, thinking we could just add a `required` flag to the function where the keys are actually optional

Reply:
I've opened an issue for it and assigned it to you: [Improve support for multiple keys in API_KEY env variable #219](https://github.com/smartcontractkit/external-adapters-js/issues/219)

Please make these few changes when you get a chance. Thx!